metricbeat/module/kafka/partition/partition.go

```diff
@@ -69,7 +71,8 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		BaseMetricSet: base,
 		broker:        broker,
 		cfg:           cfg,
-		id:            noID,
+		id:            config.ID,
```

Comment:
`ID` sets the broker id. With broker id being configured, the broker address matching will not be executed. `id` should only be configured if number of hosts being monitored is 1 (e.g. `localhost:9092` only).